### ISSUE TITLE
warn when a global is shadowed

### ISSUE THREAD

**Issue by johnfoconnor** [mpeterv/luacheck#39]
we have support for declarations of local variables shadowing previous declarations within the current scope. It would be useful to have a flag that warns when one shadows variables defined in global scope

ex

```
local type = "some string"
```

would warn as we are defining a local variable `type` which shadows the global std lib `type` function

```
type = "some string"
```

would not warn (depending on if you have configured luacheck to allow globals)

**Comment by mpeterv** [comment]
Good idea, shadowing a global can lead to same issues as shadowing locals. It can be deliberate when "localizing" a global, e.g. `local type = type`, but that case is trivial to detect. I'll implement this in the next minor version. Thanks!

**Comment by mpeterv** [comment]
Implementation delayed as it requires some refactoring: the checker needs to have information about existing globals, or else it would have to save information about all declared locals so that then the filter could leave only necessary warnings (separation into two steps is needed for caching and parallel checking.)

**Comment by mpeterv** [comment]
I've decided to not implement this because global localization is a popular and flexible idiom, it's not always in the top level scope and it doesn't always use global of the same name on rhs. So either there would be false positives or heuristics to avoid them would be too complex. I think that a better and more general way to detect this is type inference, whenever gets implemented.
